### DISCUSSION TITLE
Customization of filtering/sorting icons depending on the state

### DISCUSSION THREAD

**Discussion by ADPopov** [KevinVandy/material-react-table#270] [Ideas]
I would like to change the icons depending on the state of filtering / sorting the table. For example, change the sort icon when the state is "Unsorted", sorted "descending" and sorted "ascending". The same with filters: fill or line icon depending on the filtering state. It would be worth adding an object with the state of the table to the props of the icon component.

**Comment by KevinVandy** [comment]
There already is a way to access state wherever you need it. Please read the [MRT State Management Docs](https://www.material-react-table.com/docs/guides/table-state-management), because you have a couple of options with either the `tableInstanceRef`, or managing whatever state you need to access. 

Now currently, the `icons` prop gets memoized internally on first render, but that can change.
